tests/unit/_internal/test_strategy.py

```diff
@@ -0,0 +1,29 @@
+import os
+import typing as t
+
+import bentoml
+from bentoml._internal.runner import strategy
+from bentoml._internal.resource import get_resource as resource_get_resource
+from bentoml._internal.runner.strategy import DefaultStrategy
+
+
+class GPURunnable(bentoml.Runnable):
+    SUPPORTED_RESOURCES = ("nvidia.com/gpu",)
+
+
+def unvalidated_get_resource(x: t.Dict[str, t.Any], y: str):
+    return resource_get_resource(x, y, validate=False)
+
+
+def test_default_gpu_strategy(monkeypatch):
+    monkeypatch.setattr(strategy, "get_resource", unvalidated_get_resource)
+    DefaultStrategy.get_worker_count(GPURunnable, {"nvidia.com/gpu": 2}) == 2
+    DefaultStrategy.get_worker_count(GPURunnable, {"nvidia.com/gpu": 0}) == 0
```

Comment:
Huh, I'm surprised this works. In theory the scheduler should be finding no supported resources and falling back to using the CPU, right?

Reply:
the scheduler does that if either "system" string is specified or if the field is ignored ie only uses GPUs if available and the runner supports it. 
If you manually request for the GPU it will try and use that. There is a validation step in this case so if the GPU resources specified are unavailable, it will raise an exception. I feel this is better since it fails loudly in the case where the user did explicitly set the resource.
In the test case, I've monkey-patched it not to run validation.

what do you think?

Reply:
I'd think that if the user specified 0 their intention is to not use the resource; the strategy is supposed to check for that (https://github.com/bentoml/BentoML/blob/fdbdc140e3916bd91c6cc71cbb245a33e8105ea4/bentoml/_internal/runner/strategy.py#L66). But this is probably a strategy bug, so I think this PR is good to go.

Reply:
ohh I see what you mean. Yeah, that is strange. I'll check this again